### PR TITLE
feat (FOROME-483): add search field in filter refiner

### PR DESCRIPTION
**Pull request by YaroslavTrefilovBraveDevelopers**

### CODE REVIEW

File: src/pages/filter/ui/enum-panel.tsx [hunk 1]
```diff
@@ -20,124 +21,162 @@ export const EnumPanel = observer(
...
-      return checkedVariantNames
-    }
+    const [shouldClear, setShouldClear] = useState(false)
 
     const handleClear = () => {
-      updateSubattributes()
-      datasetStore.fetchDsStatAsync()
+      const localSelectedFilters = JSON.parse(
+        JSON.stringify(filterStore.selectedFilters),
+      )
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/enum-panel.tsx:72]: better to use cloneDeep from `'lodash/cloneDeep'`. It's faster and keeps the original type. While JSON.parse/stringify always returns `any`

File: src/pages/filter/ui/enum-panel.tsx [hunk 2]
```diff
@@ -20,124 +21,162 @@ export const EnumPanel = observer(
...
           'OR',
-          variantNameList,
+          nameVariantList,
         ],
       ])
+      setCurrentPage(0)
+
+      const localSelectedFilters = JSON.parse(
+        JSON.stringify(filterStore.selectedFilters),
+      )
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/enum-panel.tsx:103]: The same as above

File: src/pages/filter/ui/enum-panel.tsx [hunk 3]
```diff
@@ -20,124 +22,159 @@ export const EnumPanel = observer(
...
+    }, [groupItemName])
 
-      return checkedVariantNames
-    }
+    const [shouldClear, setShouldClear] = useState(false)
 
     const handleClear = () => {
-      updateSubattributes()
-      datasetStore.fetchDsStatAsync()
+      cloneDeep
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/enum-panel.tsx:71]: forgot to delete?

File: src/pages/filter/ui/enum-panel.tsx [hunk 4]
```diff
@@ -1,16 +1,18 @@
 import { ReactElement, useEffect, useState } from 'react'
+import { cloneDeep } from 'lodash'
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/enum-panel.tsx:2]: avoid importing from `lodash`, prefer to import from `lodash/%functionName%`.
The problem is, when you import something from lodash, you import the whole library, and it's enormously large. On the opposite, when you import a particular function from lodash like this - `import cloneDeep from 'lodash/cloneDeep'`, you import the only function you need. And this makes the bundle way lighter.

**Review by AndreyUstyumenko — APPROVED** [review]

**Review by Lapk1n — APPROVED** [review]

**Pull request merged**